### DISCUSSION TITLE
error building flex app w intellij on mac

### DISCUSSION THREAD

**Discussion by studiomaxer3d** [airsdk/Adobe-Runtime-Support#1737] [Q&A]
I've taken a break from an app I built a while back and recently had to come back in to do some updates. 

Flex mobile app. I put together the air sdk over the same flex sdk I've been using for years. Now I'm getting this error when trying to build.

 unable to load SWC ._playerglobal_rb.swc: .../flex_sdk4.16.1_air33.1.1.779/frameworks/locale/en_US/._playerglobal_rb.swc is not a SWC file.

**Comment by ajwfrost** [comment]
Which flavour of AIR SDK did you use? The SDK that's intended for overlay onto a Flex SDK does not include the `frameworks/locale` folder... but then, I'm not sure what would be asking for `._playerglobal_rb.swc` as the files there are actually just called `playerglobal_rb.swc` - is there any info about where this is coming from? Maybe check the `frameworks\air-config.xml` file or whatever config file you're loading, to see if it's referenced there?

**Reply by studiomaxer3d** [reply]
I grabbed the latest air from harman. It must be macOS because even my old SDKs won't work and I get the same issue on another mac with a fresh Intellij install.

**Reply by studiomaxer3d** [reply]
I get this issue too and if I change it to 33.1.1 it gets mad
![image](https://user-images.githubusercontent.com/11936072/156506870-a30fbdb3-802a-44d1-99a7-ce6a7440ad5d.png)

**Reply by ajwfrost** [reply]
There are two different types of the AIR SDKs, for each operating system. If you had downloaded the full AIR SDK with the new compiler, it would have overwritten some of the Flex SDK components that would still be needed if you're building MXML. So you need the one under the heading "AIR SDK for Flex Developers - to be overlaid onto a Flex SDK"...

Never seen that warning before about the AIR SDK version being 33.1.1 and then not matching the app descriptor. Yes 33.1 would be needed there but what IDE is this that's picking up "33.1.1" from somewhere? (most likely from the air-sdk-description.xml file in the root of the SDK folder, you can perhaps try editing this?)

thanks

**Reply by ajwfrost** [reply]
.. oh you'd already said, it's IntelliJ. Curious as we've not seen that with the same description file... but perhaps edit that XML and change the 'version' tag to just be 33.1?

**Reply by studiomaxer3d** [reply]
Ok, yeah I did download the right AIR SDK. I'm perplexed. I was in the IDE a few months ago making updates and everything was fine. Now I've got some users in need of an update and I have this weird error.

**Reply by studiomaxer3d** [reply]
Maybe I need to build a new Flex SDK, but the apache flex installer that I used to use is now not working.
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/11936072/156574587-72dd4f61-5d39-4b24-855a-26d6fba0954c.png">

**Reply by 2jfw** [reply]
For overlaying **AIR SDK for Flex Developers** _onto_ Flex SDK we always use the Flex SDK "binaries" from here (because Apache's Flex SDK Installer does not seem to work properly anymore):

[flex.apache.org/download-binaries.html](https://flex.apache.org/download-binaries.html)

We are on Windows but there is also the **MAC version** linked:
Mac OS X Binaries : [apache-flex-sdk-4.16.1-bin.tar.gz](https://www.apache.org/dyn/closer.lua/flex/4.16.1/binaries/apache-flex-sdk-4.16.1-bin.tar.gz)

EDIT:
So, basically the steps are download both SDKs (make sure to get the right AIR SDK for Flex) and then copy the content from AIR SDK over the Flex SDK and overwrite all files.

**Reply by studiomaxer3d** [reply]
I tried this and I'm still getting

unable to load SWC ._playerglobal_rb.swc: .../flex_sdk4.16.1_air33.1.1.779/frameworks/locale/en_US/._playerglobal_rb.swc is not a SWC file.

I'm ready to lose my marbles.

**Reply by ajwfrost** [reply]
So, something must be trying to reference that (with some weird temporary name..? or path before the swc?)

When it's compiling, you're presumably getting a "loading config" message, do you know which file it is? or is that your own version? I don't actually know what would make it look in that particular path .. as there is no reference anywhere to the ..._rb.swc files (afaik), though I may be wrong!

I'm wondering if the simplest workaround would be to go into that folder on your machine, and copy the file that is there (`playerglobal_rb.swc`) so that it also is present with the requested name (`._playerglobal_rb.swc`).

Bit weird though...!

**Reply by studiomaxer3d** [reply]
So, after playing around I got it to be asking about advancedgrids_rb.swc. The thing is when I create the file manually, macos is acting like its already there. I'm displaying hidden files and it's not. macos won't let me create the file.

**Reply by studiomaxer3d** [reply]
so, long story short I think the problem may be related to running my project off an external drive. I moved to a different macbook with more hard drive space and I'm moving past this issue and just trying to get everything else set up.
